client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,282 @@
+        this.searchRequest.preference(preference);
+    }
+
+    /**
+     * Get the preference to execute the search.
+     */
+    public String getPreference() {
+        return this.searchRequest.preference();
+    }
+
+    /**
+     * Specifies what type of requested indices to ignore and how to deal with indices wildcard expressions.
+     */
+    public void setIndicesOptions(IndicesOptions indicesOptions) {
+        this.searchRequest.indicesOptions(indicesOptions);
+    }
+
+    /**
+     * Get the indices Options.
+     */
+    public IndicesOptions getIndicesOptions() {
+        return this.searchRequest.indicesOptions();
+    }
+
+    /**
+     * The search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public void setSearchType(SearchType searchType) {
+        this.searchRequest.searchType(searchType);
+    }
+
+    /**
+     * Get the search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public SearchType getSearchType() {
+        return this.searchRequest.searchType();
+    }
+
+    /**
+     * Sets if this request should allow partial results. (If method is not called,
+     * will default to the cluster level setting).
+     */
+    public void setAllowPartialSearchResults(boolean allowPartialSearchResults) {
+        this.searchRequest.allowPartialSearchResults(allowPartialSearchResults);
+    }
+
+    /**
+     * Gets if this request should allow partial results.
+     */
+    public Boolean getAllowPartialSearchResults() {
+        return this.searchRequest.allowPartialSearchResults();
+    }
+
+    /**
+     * Sets the number of shard results that should be reduced at once on the coordinating node. This value should be used as a protection
+     * mechanism to reduce the memory overhead per search request if the potential number of shards in the request can be large.
+     */
+    public void setBatchedReduceSize(int batchedReduceSize) {
+        this.searchRequest.setBatchedReduceSize(batchedReduceSize);
+    }
+
+    /**
+     * Gets the number of shard results that should be reduced at once on the coordinating node.
+     * This defaults to 5 for {@link SubmitAsyncSearchRequest}.
+     */
+    public int getBatchedReduceSize() {
+        return this.searchRequest.getBatchedReduceSize();
+    }
+
+    /**
+     * Sets if this request should use the request cache or not, assuming that it can (for
+     * example, if "now" is used, it will never be cached). By default (not set, or null,
+     * will default to the index level setting if request cache is enabled or not).
+     */
+    public void setRequestCache(Boolean requestCache) {
+        this.searchRequest.requestCache(requestCache);
+    }
+
+    /**
+     * Gets if this request should use the request cache or not.
+     * Defaults to `true` for {@link SubmitAsyncSearchRequest}.
+     */
+    public Boolean getRequestCache() {
+        return this.searchRequest.requestCache();
+    }
```

Comment:
how about maxConcurrentShardRequests? Was it left out on purpose?

Reply:
Good one, I think that parameter is also missing from the regular RequestConverters.

Reply:
I pulled adding this parameter (and the previously mentioned pre_filter_shard_size) to the regular RequestConverters into https://github.com/elastic/elasticsearch/pull/53745 to get this in independently of this PR.

Reply:
sounds good thanks

Reply:
Cherry-picked that PR and added support for max_concurrent_shard_requests